src/MizarData/include/MizarData/MizarPairedData.h

```diff
@@ -141,6 +142,40 @@ class MizarPairedDataTmpl {
   }
 
  private:
+  struct Frame {
+    TimestampNs start;
+    TimestampNs end;
+  };
+
+  // Op is a `void` callable that takes `Accumulator&` and `Frame`.
+  template <typename Accumulator, typename Op>
+  [[nodiscard]] Accumulator ReduceOverFrames(FrameTrackId frame_track_id,
+                                             RelativeTimeNs min_relative_time,
+                                             RelativeTimeNs max_relative_time, Op&& op) const {
+    Accumulator accumulator;
+    const auto [min_timestamp_ns, max_timestamp_ns] =
+        RelativeToAbsoluteTimestampRange(min_relative_time, max_relative_time);
+    const std::vector<TimestampNs> frame_starts =
+        GetFrameStarts(frame_track_id, min_timestamp_ns, max_timestamp_ns);
+    if (frame_starts.size() < 2) return accumulator;
+
+    for (size_t i = 0; i + 1 < frame_starts.size(); ++i) {
+      std::invoke(op, accumulator, Frame{frame_starts[i], frame_starts[i + 1]});
+    }
+    return accumulator;
+  }
+
+  [[nodiscard]] RelativeTimeNs FrameActiveInvocationTime(const absl::flat_hash_set<TID>& tids,
+                                                         Frame frame) const {
+    const uint64_t callstack_count =
+        std::transform_reduce(std::begin(tids), std::end(tids), uint64_t{0}, std::plus<>(),
+                              [this, &frame](const TID tid) {
```

Comment:
```suggestion
                              [this, &frame](const TID tid) -> uint64_t {
```

Maybe that helps reading a bit?

For some weird reason, I was first under the impression that the `return CallstackSamplesCount(tid, frame.start, frame.end);` constructs something that is a `CallstackSamplesCount` and that this structure has a `+` operator defined.

Maybe also, rename the `CallstackSamplesCount` to `ComputeCallstackSampleCount` or similar?

Reply:
With this -- I fully agree. `CallstackSamplesCount` is a terrible name for a function. At the very least, it's not a verb. Let's call it `CountCallstackSamples`. This makes it clear it returns an integral type and we don't have to explicitly specify the return type of the lambdas.

Reply:
Yeah, I think this already makes it a lot better.